### ISSUE TITLE
Unexpected behavior pg8000

### ISSUE THREAD

**Issue by KULINAR847** [tlocke/pg8000#93]
https://github.com/KULINAR847/ChudoBagPg8000

**Comment by tlocke** [comment]
Hi @KULINAR847, I've had a look but what would really help us is if you could provide a [Minumum Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example). What we're looking for is the smallest bit of code that will cause the bug to appear.

**Comment by KULINAR847** [comment]
Hi Tony! I made a simple example. I committed it! And corrected README.md
https://github.com/KULINAR847/ChudoBagPg8000

Best Wishes,
Ivan Konoplev

**Comment by tlocke** [comment]
I think we need to do a bit more narrowing down to isolate the bug in pg8000. My suggestion would be to put everything in one file, and have the SQL as literal strings. The thing is that you're doing quite a bit of processing before the SQL is executed, but we want to end up with an example where literal strings are being executed by pg8000.

**Comment by KULINAR847** [comment]
Collected everything into one file and SQL as strings. Corrected README.md.
https://github.com/KULINAR847/ChudoBagPg8000
<https://github.com/KULINAR847/ChudoBagPg8000>

**Comment by tlocke** [comment]
Thanks that's a lot easier to understand now. So we've got:

```
import os
import pg8000

ddl = r'''
DO $$ BEGIN
CREATE OR REPLACE FUNCTION public.function (
  out code integer
)
RETURNS integer AS
$body$
try:
    pass
except urllib.error.HTTPError as e1:
    plpy.execute("SELECT public.errwrite('getRecipe', 12, ARRAY['%s','%s','%s'])" % (e1.code, e1.reason.replace("\\\\","\\").replace("'","\""), e1.read().decode('utf-8').replace("\\\\","\\").replace("'","\"")))
    return [123, e1.reason]
    
except Exception as e2:
    #plpy.execute("SELECT public.errwrite('getRecipe', 13, ARRAY['%s'])" % str(e2).replace("\\\\","\\").replace("'","\""))
    plpy.execute("SELECT public.errwrite('getRecipe', 13, ARRAY['%s'])" % str(e2).replace("\\\\","\\").replace("'","\""))
    return [1232, str(e2)]
$body$
LANGUAGE 'plpython3u'
VOLATILE
CALLED ON NULL INPUT
SECURITY INVOKER
COST 100;
ALTER FUNCTION public.function (out code integer)
  OWNER TO postgres;
END$$;
'''

if __name__ == '__main__':
    try:
        ddl = ddl.replace(' % ', ' %% ')

        conn = pg8000.connect(user='postgres', password='pass', database='test_db', host='localhost', port=5432, timeout=0.500)
        cursor = conn.cursor()
        cursor.execute(ddl)

        conn.commit()
        cursor.close()
        conn.close()

    except Exception as e:
        print(e)
        input()
```

So looking at this I realized that pg8000 is parsing the query and potentially rewriting it, even though there aren't any parameters. I've fixed this now and it's in version 1.23.0. So now, in your code the line:

        ddl = ddl.replace(' % ', ' %% ')

should be removed for use with version 1.23.0. Let me know if you still have a problem after trying with version 1.23.0. Also, thanks very much for the bug reporting, it's revealed a big problem with pg8000!

**Comment by KULINAR847** [comment]
Yes, all work fine.
Closes: #93
